encoding/csv/stream.ts

```diff
@@ -0,0 +1,51 @@
+import { Buffer, BufReader } from "../../io/buffer.ts";
+import { defaultReadOptions, readRecord } from "./_io.ts";
+
+function createTransformStream(
+  options?: CSVStreamOptions,
+): TransformStream<Uint8Array, Array<string>> {
+  let lineIndex = 0;
+  const buffer = new Buffer();
+  const bufReader = BufReader.create(buffer);
+  return new TransformStream<Uint8Array, Array<string>>({
+    async transform(chunk, controller) {
+      buffer.writeSync(chunk);
+      while (!buffer.empty() || bufReader.buffered() > 0) {
+        const record = await readRecord(lineIndex, bufReader, options);
+        if (record === null && buffer.empty()) {
+          controller.terminate();
+          return;
+        }
+
+        lineIndex++;
+        if (record && record.length > 0) {
+          controller.enqueue(record);
+        }
+      }
+    },
+  });
+}
+
+export interface CSVStreamOptions {
+  separator?: string;
+  comment?: string;
+}
+
+export class CSVStream {
```

Comment:
You could rather make `CSVStream` extend `TransformStream`, then the only thing you have to do is `super()` with the transformstream options; and the variables you defined can just be private properties.

This isnt necessary, but imo just a lot cleaner & direct